lib/rspec/core/option_parser.rb

```diff
@@ -140,8 +140,18 @@ def parser(options)
           options[:full_backtrace] = true
         end
 
-        parser.on('-c', '--[no-]color', '--[no-]colour', 'Enable color in the output.') do |o|
-          options[:color] = o
+        parser.on('-c', '--color', '--colour', 'Deprecated: enable color in the output.') do |_o|
+          options[:color] = true
+        end
+
+        parser.on('--force-color', '--force-colour', 'Force the output to be in color.') do |_o|
+          options[:force_color] = true
+          options.delete(:no_color)
```

Comment:
Previously, we used the built-in `OptionParser` behavior where passing `--no-color` would just unset the `:color` option, since there were only two states: `:color` set, or default. But now there are three needed states:

1. Default behavior: show color iff tty
2. Show color regardless of tty
3. Do not show color, regardless of tty

So we need a way to indicate any of those states to the configuration. I opted for having a `:force_color` option and `:no_color` option, so that they directly correspond to the two options.

So then the question is, what to do if both are set? The previous behavior of the system was that whichever flag was set last wins. But now that they are two different options, setting both flags would just set both options to true. And there would be no way for the configuration to tell the order they were set in: it just receives the two booleans.

So, to preserve the behavior of the last-flag-winning, whenever `:force_color` or `:no_color` is set, I unset the opposite flag.

Thoughts?